var/spack/repos/builtin/packages/py-nltk/package.py

```diff
@@ -13,11 +13,28 @@ class PyNltk(PythonPackage):
 
     version('3.5', sha256='845365449cd8c5f9731f7cb9f8bd6fd0767553b9d53af9eb1b3abf7700936b35')
 
+    variant('data', default=False, description='Download the NLTK data')
+
     depends_on('python@3.5:', type=('build', 'run'))
-    depends_on('py-setuptools', type='build')
+    depends_on('py-setuptools', type=('build', 'run'))
     depends_on('py-click', type=('build', 'run'))
     depends_on('py-regex', type=('build', 'run'))
     depends_on('py-tqdm', type=('build', 'run'))
+    depends_on('py-importlib-metadata', type=('build', 'run'))
+
+    def setup_run_environment(self, run_env):
+        if '+data' in self.spec:
+            run_env.prepend_path("NLTK_DATA", self.prefix.data)
+
+    @run_after('install')
+    def install_data(self):
+        if '+data' in self.spec:
+            mkdirp(prefix.data)
+            python('-m',
+                   'nltk.downloader',
+                   '-d',
+                   prefix.data,
+                   'all')
```

Comment:
I assume this requires internet access? I wonder if there is a way we could add the data as resources.

Reply:
Gave that a go.